common/flags/tests/FlagsTest.java

```diff
@@ -133,14 +134,16 @@ public void firstArgsThenFlagsTest() {
   }
 
   @Test
-  public void printUsageTest() {
+  public void printUsageTest() throws IOException {
```

Comment:
How come this compiled before you added `throws IOException`?
I thought if code can throw an exception, it won't compile unless you declare it, as you now did.

Reply:
It was compiling successfully until I added `try-with-resources` construction for `ByteArrayOutputStream`. After that, I needed to add an exception to the method signature or catch block. Is `try-with-resources` construction redundant for a unit test?

Reply:
@val-fed 99% of the time, `try-with-resources` is _very_ good to use regardless of whether the code is test code or production code. :)

In this particular case, `try-with-resources` is not strictly necessary, because IIRC, `ByteArrayOutputStream.close()` (which is what gets called by the `try-with-resources` block when the block is exited) doesn't actually do anything, since `ByteArrayOutputStream` doesn't manage any external I/O resources that need to be explicitly closed.

So long story short, in this case, `try-with-resources` is a bit redundant. :)

Reply:
@jbduncan but then any dev (also those just reading the code) should know that about `ByteArrayOutputStream`, otherwise it looks like a mistake.
This was done based on a linter, right @val-fed ?
If so, then perhaps the linter could somehow be smarter about that.